src/main/java/com/robinpowered/sdk/credential/AccessTokenCredential.java

```diff
@@ -23,10 +23,18 @@
      */
 
     public AccessTokenCredential(String token) {
+        if (token == null) {
+            throw new IllegalArgumentException("Credentials cannot have a null values.");
+        }
+
         this.token = token;
     }
 
     public AccessTokenCredential(Token token) {
+        if (token == null || token.getAccessToken() == null) {
```

Comment:
Again, shouldn't you validate this in the `Token` constructor itself, versus here?